migrator/migration-steps.js

```diff
@@ -242,6 +241,95 @@ async function updateUnbrandedLayouts (dir) {
   return results.flat()
 }
 
+async function upgradeIfPossible (filePath, matchFound) {
+  if (matchFound) {
+    return true
+  } else {
+    const reporter = await addReporter(`Update ${filePath}`)
+    const fullPath = path.join(projectDir, filePath)
+    const filename = fullPath.split(path.sep).pop()
+    if (filename === 'application.js') {
+      await upgradeApplicationJs(fullPath, reporter)
+      return true
+    }
+    await reporter(false)
+    return false
+  }
+}
+
+async function upgradeApplicationJs (fullPath, reporter) {
+  const textToReplace = [
+    {
+      originalText: '/* global $ */',
+      replacementText:
+        `//
+// For guidance on how to add JavaScript see:
+// https://prototype-kit.service.gov.uk/docs/adding-css-javascript-and-images
+//`
+    },
+    {
+      originalText:
+        `// Warn about using the kit in production
+if (window.console && window.console.info) {
+window.console.info('GOV.UK Prototype Kit - do not use for production')
+}`
+    },
+    {
+      originalText: '$(document).ready(function () {',
+      replacementText: 'window.GOVUKPrototypeKit.documentReady(() => {'
+    },
+    {
+      originalText: 'window.GOVUKFrontend.initAll()',
+      replacementText: '  // Add JavaScript here'
+    }
+  ]
+  const fileBuffer = await fsp.readFile(fullPath)
+  const fileContent = fileBuffer.toString()
+  const fileLines = fileContent.split('\n')
+  const newContentLines = []
+  let lastPos = 0
```

Comment:
I'm reading through the code and wondering whether `lastPos` should be `currentPos` - it seems to be holding the current line index but the variable name (particularly in the same context as `startPos` sounds like it's equivalent to `endPos`.